parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -212,49 +339,82 @@ try
-                         -AgentCertificate $global:AgentCertificate
+            -KubeDir $global:KubeDir `
+            -MasterFQDNPrefix $MasterFQDNPrefix `
+            -MasterIP $MasterIP `
+            -AgentKey $AgentKey `
+            -AgentCertificate $global:AgentCertificate
+
+        if ($global:EnableHostsConfigAgent) {
+             Write-Log "Starting hosts config agent"
+             New-HostsConfigService
+         }
 
         Write-Log "Create the Pause Container kubletwin/pause"
-        New-InfraContainer -KubeDir $global:KubeDir
+        if ($global:EnableTelemetry) {
+            $infraContainerTimer = [System.Diagnostics.Stopwatch]::StartNew()
+        }
+        New-InfraContainer -KubeDir $global:KubeDir -ContainerRuntime $global:ContainerRuntime
+        if ($global:EnableTelemetry) {
+            $infraContainerTimer.Stop()
+            $global:AppInsightsClient.TrackMetric("New-InfraContainer", $infraContainerTimer.Elapsed.TotalSeconds)
+        }
 
-        if (-not (Test-ContainerImageExists -Image "kubletwin/pause")) {
+        if (-not (Test-ContainerImageExists -Image "kubletwin/pause" -ContainerRuntime $global:ContainerRuntime)) {
             Write-Log "Could not find container with name kubletwin/pause"
-            $o = docker image list
-            Write-Log $o
+            if ($useContainerD) {
+                $o = ctr -n k8s.io image list
+                Write-Log $o
+            } else {
+                $o = docker image list
+                Write-Log $o
+            }
             throw "kubletwin/pause container does not exist!"
         }
 
         Write-Log "Configuring networking with NetworkPlugin:$global:NetworkPlugin"
 
         # Configure network policy.
+        Get-HnsPsm1 -HNSModule $global:HNSModule
+        Import-Module $global:HNSModule
+
         if ($global:NetworkPlugin -eq "azure") {
             Write-Log "Installing Azure VNet plugins"
             Install-VnetPlugins -AzureCNIConfDir $global:AzureCNIConfDir `
-                                -AzureCNIBinDir $global:AzureCNIBinDir `
-                                -VNetCNIPluginsURL $global:VNetCNIPluginsURL
+                -AzureCNIBinDir $global:AzureCNIBinDir `
+                -VNetCNIPluginsURL $global:VNetCNIPluginsURL
+
             Set-AzureCNIConfig -AzureCNIConfDir $global:AzureCNIConfDir `
-                               -KubeDnsSearchPath $global:KubeDnsSearchPath `
-                               -KubeClusterCIDR $global:KubeClusterCIDR `
-                               -MasterSubnet $global:MasterSubnet `
-                               -KubeServiceCIDR $global:KubeServiceCIDR `
-                               -VNetCIDR $global:VNetCIDR `
-                               -TargetEnvironment $TargetEnvironment
+                -KubeDnsSearchPath $global:KubeDnsSearchPath `
+                -KubeClusterCIDR $global:KubeClusterCIDR `
+                -MasterSubnet $global:MasterSubnet `

```

Comment:
we shouldn't need this masterSubnet

Reply:
Windows will add this subnet into the Exception list. It may block the connections from Windows nodes if we remove it.

Reply:
It's a subnet of vnetcidr, which is already in the exception list.

Reply:
So we can remove this.

Reply:
Could we refine the code in next PR but not in this PR?